packages/react-components/src/components/data-providers/tsp-data-provider.ts

```diff
@@ -94,6 +100,11 @@ export class TspDataProvider {
             }
         }
         const arrows = await this.getArrows(ids, viewRange, resolution);
+
+        rows.forEach(row => {
+            row.annotations = row.annotations.concat(rangeEvents);
```

Comment:
annotations in that span the whole view should not be stored in each row. They are an additional layer to be drawn on the chart. I think, there should be a new layer in the timeline-chart library.

Also, I noticed that the view gets very unresponsive. in my test trace I have only 5 such annotations and that should not impact the drawing performance. Could it be that in the timeline-chart each such marker is draw for each row>

Reply:
Added a new layer for lost event. Thank you for the feedback and suggestion!!